Impossible-Agent-Suite/flash-loan/src/agent.ts

```diff
@@ -0,0 +1,129 @@
+import {
+  Finding,
+  HandleTransaction,
+  TransactionEvent,
+  FindingSeverity,
+  FindingType,
+  getEthersProvider,
+} from 'forta-agent';
+
+const { ethers } = require('ethers');
+
+import LRU from 'lru-cache';
+
+const cache: LRU<string, boolean> = new LRU<string, boolean>({ max: 10000 });
+
+//let SWAP_FACTORY_ADDRESS = '0x918d7e714243f7d9d463c37e106235dcde294ffc';
+let SWAP_FACTORY_ADDRESS = '0x5C69bEe701ef814a2B6a3EDD4B1652CB9cc5aA6f';
+
+export const SWAP_FACTORY_ABI = [
+  'event PairCreated(address indexed token0, address indexed token1, address pair, uint)',
+  'function getPair(address tokenA, address tokenB) external view returns (address pair)',
+];
+
+export const PAIR_SWAP_ABI = [
+  'event Swap(address indexed sender, uint amount0In, uint amount1In, uint amount0Out, uint amount1Out, address indexed to)',
+  'function swap(uint amount0Out, uint amount1Out, address to, bytes calldata data)',
+  'function token0() external view returns (address)',
+  'function token1() external view returns (address)',
+];
+
+// Receives the address of a contract with same ABI as PAIR_SWAP_ABI and returns the factory address
+export const checkFromFactory = async (swapContract: string) => {
+  let isFromFactory;
+  // If the contract exists in the cache
+  if(cache.has(swapContract)) {
+    // Get data from cache
+    isFromFactory = cache.get(swapContract);
+  } else {
+  // Otherwise if the cache does contain data for `swapContract`
+    try {
+      // Query the smart contract for its tokens
+      const provider = getEthersProvider();
+      const contractInterface = new ethers.Contract(swapContract, PAIR_SWAP_ABI, provider);
+      const token0 = await contractInterface.token0();
+      const token1 = await contractInterface.token1();
+      // Query the factory to see if the token pairs point to the contract address
+      const factoryInterface = new ethers.Contract(SWAP_FACTORY_ADDRESS, SWAP_FACTORY_ABI, provider);
+      const pairAddress = await factoryInterface.getPair(token0, token1);
+      // If the returned pair matches `swapContract` then add to the cache
+      if(pairAddress.toLowerCase() == swapContract.toLowerCase()) {
+        cache.set(swapContract, true);
+        isFromFactory = true;
+      } else {
+        cache.set(swapContract, false);
+        isFromFactory = false;
+      }
+    } catch(error) {
+      // If the query fails
+      cache.set(swapContract, false);
+      isFromFactory = false;
+    }
+  }
+  return isFromFactory;
+}
+
+export const provideHandleTransaction = (
+  address: string,
```

Comment:
this variable is not being used. Please change its name, and pass it to the `checkFromFactory` function call in Line 98 in order to avoid using the global variable `SWAP_FACTORY_ADDRESS` in Line 47

Reply:
This will be addressed in the following commit.